src/textual/widgets/_tabs.py

```diff
@@ -0,0 +1,440 @@
+            label: The label to use in the tab. May be a str or a Text object.
+            id: Optional ID for the widget.
+
+        """
+        self.label = Text.from_markup(label) if isinstance(label, str) else label
+        if id is None:
+            id = f"tab-{self.label_text.lower().replace(' ', '-')}"
+        super().__init__(id=id)
+        self.update(label)
+
+    @property
+    def label_text(self) -> str:
+        """Undecorated text of the label."""
+        return self.label.plain
+
+    def _on_click(self):
+        """Inform the message that the tab was clicked."""
+        self.post_message(self.Clicked(self))
+
+
+class Tabs(Widget, can_focus=True):
+    """A row of tabs."""
+
+    DEFAULT_CSS = """
+    Tabs {
+        width: 100%;
+        height:3;
+    }
+    Tabs > #tabs-scroll {
+        overflow: hidden;
+    }
+    Tabs #tabs-list {
+       width: auto;
+       min-height: 2;
+    }
+    Tabs #tabs-list-bar, Tabs #tabs-list {
+        width: auto;
+        height: auto;
+        min-width: 100%;
+        overflow: hidden hidden;
+    }
+    Tabs:focus .underline--bar {
+        background: $foreground 20%;
+    }
+    """
+
+    BINDINGS: ClassVar[list[BindingType]] = [
+        Binding("left", "previous_tab", "Previous tab", show=False),
+        Binding("right", "next_tab", "Next tab", show=False),
+    ]
+
+    class TabActivated(Message):
+        """Sent when a new tab is activated."""
+
+        def __init__(self, tab: Tab | None) -> None:
+            self.tab = tab
+            super().__init__()
+
+        def __rich_repr__(self) -> rich.repr.Result:
+            yield self.tab
+
+    active: reactive[str] = reactive(str, init=False)
+
+    def __init__(
+        self,
+        *tabs: Tab | TextType,
+        active: str | None = None,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Construct a Tabs widget.
+
+        Args:
+            *tabs: Positional argument should be explicit Tab objects, or a str or Text.
+            active (_type_, optional): ID of the tab which should be active on start.
+            name: Optional name for the input widget.
+            id: Optional ID for the widget.
+            classes: Optional initial classes for the widget.
+            disabled: Whether the input is disabled or not.
+        """
+        self._tabs_counter = 0
+        add_tabs = [
+            (
+                Tab(tab, id=f"tab-{self._new_tab_id}")
+                if isinstance(tab, (str, Text))
+                else tab
+            )
+            for tab in tabs
+        ]
+        super().__init__(
+            name=name,
+            id=id,
+            classes=classes,
+            disabled=disabled,
+        )
+        self._tabs = add_tabs
+        self._first_active = active
+
+    @property
+    def _new_tab_id(self) -> int:
+        """Get the next tab id in a sequence."""
```

Comment:
```suggestion
        """Next tab ID in a sequence."""
```

Also, what's "a sequence"? "The next available tab ID"?

Reply:
Exactly.